### DISCUSSION TITLE
How to make dirtyFields reflect form values changes ?

### DISCUSSION THREAD

**Discussion by nphdiem** [react-hook-form/react-hook-form#9117] [Q&A] [answered]
Hi, I am working on form validation and I encounter an issue that I can not get the correct `dirtyFields` after I use `append`.
You can see code example here:  https://codesandbox.io/s/register-forked-vlb06u?file=/src/RegisterForm.tsx

Does anyone know the reason and how I can fix it ?
Thanks in advance.

**Comment by bluebill1049** [comment, marked as answer]
form state update will be flushed in the next render not withing the action.

```tsx
const handleValidation = () => {
   append({ data: 'test' })
   console.log(formState.dirtyFields); // ❌
};

useEffect(() =>  {
  console.log(formState.dirtyFields); // ✅
}, [formState ])
```

**Reply by nphdiem** [reply]
I got it. Thank you :D
